woocommerce/woocommerce-gutenberg-products-block
# Merge ProductPrice atomic block and component

Part of #1967.

This PR:
* Updates the `ProductPrice` component so it can handle price ranges and loading states, in order to have feature-parity with the Product Price atomic block.
* Moves the `ProductPrice` component out of the `cart-checkout` subdirectory.
* Updates the Product Price atomic block so internally it uses the `ProductPrice` component, some benefits of this:
  * The markup is centralized in a single place. So any enhancement we do, will affect Cart & Checkout + All Products & Single Product.
  * Similarly, styles can all be in a single file too. Preventing things like #3048.

### How to test the changes in this Pull Request:

Testing basically means verifying there are no regressions:

1. Load the Cart and Checkout blocks and verify product prices (default prices and on sale prices) are still rendered correctly.
<img src="https://user-images.githubusercontent.com/3616980/91466321-3d9eb500-e88f-11ea-9a69-554c149a4163.png" alt="" width="187" />
2. Create a page with the Product Price block for three products (one with a default price, another one on sale and another one with a price range--grouped product). Verify prices display correctly and you can style them in the editor.
<img src="https://user-images.githubusercontent.com/3616980/91466346-47281d00-e88f-11ea-910a-73cb07114349.png" alt="" width="127" />
3. Add the All Products block in a page and verify there are no regressions with the Product Price block.
